.azure-pipelines/build-template.yml

```diff
@@ -84,38 +116,39 @@ jobs:
       artifact: ${{ parameters.swss_common_artifact_name }}
       runVersion: 'latestFromBranch'
       runBranch: 'refs/heads/$(BUILD_BRANCH)'
-      path: $(Build.ArtifactStagingDirectory)/download
+      allowFailedBuilds: ${{ parameters.allow_failed_artifact }}
```

Comment:
Why do we need to allow failed builds here?

Reply:
The default value for allow_failed_artifact is false. So perhaps that is ok?

Reply:
Looks like it's set to True in `azure-pipeline-dash.yml` below

Reply:
Because we are using the branch, dash, of sonic-swss-common which also leverage the sairedis code in Azp: https://dev.azure.com/mssonic/build/_build/results?buildId=166787&view=logs&jobId=61b7eb0d-78fa-5af5-6df0-b6f82d3891a8&j=61b7eb0d-78fa-5af5-6df0-b6f82d3891a8&t=70f86778-6fb4-5097-2f4f-43a5d7a20e70 . But in the repo, sonic-sairedis, there is no related branch named 'dash'.
I don't think we can directly use the master branch of sonic-swss-common, because other artifacts from DASH like 'syncd' and so on were using  a specified commit of sonic-swss-common what was listed in sonic-buildimage.

Reply:
As per https://github.com/sonic-net/sonic-swss/pull/2501#discussion_r1012389884, we should be using the master branch of swss common. In this case, I don't think we still need to allow failed builds.